### DISCUSSION TITLE
PyAnnote error despite MacOS 12.3+. (NoScribe 0.4.0)

### DISCUSSION THREAD

**Discussion by buhro** [kaixxx/noScribe#43] [General]
I got following error in identification the speakers: 

"PyAnnote error:   The MPS backend is supported on MacOS 12.3+.Current OS version can be queried using `sw_vers`"

"Fehler im 2. Schritt - Erkennung der Sprecher:innen."

I have McOS Monterey 12.7.2, intel, so it should be enough. I used NoScribe 0.4.0 because 0.4.3 don't open on my intel Mac -> here ist this problem: https://github.com/kaixxx/noScribe/discussions/42

After deactivating the identification of speakers, everything works fine. 

Thanks for supporting!

**Comment by gernophil** [comment]
That's a problem with MPS recognition (the Apple GPU acceleration framework for machine learning). Problem is, that I would need to load a sepcific Python package to check your specific system combination, but if I do that another part of the program crashes. You have macOS bigger then 12.7.2, but you don't seem to have a MPS compatible GPU. There should be a fallback for this, but it seems not to work for you for some reason. To set this manually go to the file `~/Library/Application Support/noScribe/config.yml` and change the line `pyannote_xpu: mps` to `pyannote_xpu: cpu`.

**Comment by buhro** [comment]
Thank you for answering! But i can't solve this problem, because there isn't such a file "config.yml". In the Application Support of noScribe is only a file for the log: 

<img width="758" alt="Bildschirmfoto 2024-01-29 um 10 28 38" src="https://github.com/kaixxx/noScribe/assets/157977717/c53e3dea-0b9a-4875-9202-5399fad7ec40">

**Reply by gernophil** [reply]
That's a bit weird. It should be there, since noScribe did run successfully without speaker identification. And it should be in [this folder](https://pypi.org/project/appdirs/).
Try to simply create a plain text file and name it `config.yml` then and simply put `pyannote_xpu: cpu` int there :).
